public/app/features/dashboard/save_modal.ts

```diff
@@ -50,8 +50,17 @@ const template = `
     </div>
 
     <div class="gf-form-button-row text-center">
-      <button type="submit" class="btn btn-success" ng-disabled="ctrl.saveForm.$invalid">Save</button>
-      <a class="btn btn-link" ng-click="ctrl.dismiss();">Cancel</a>
+      <button
+        id="saveBtn"
+        type="submit"
+        class="btn btn-success"
+        ng-class="{'btn-success--processing': ctrl.isSaving}"
+        ng-disabled="ctrl.saveForm.$invalid"
```

Comment:
Did you try `ng-disabled="ctrl.saveForm.$invalid || ctrl.isSaving"`? Haven't tried it, but if that works I think you can remove the extra spans below and some logic in the controller.